### ISSUE TITLE
PSR2 standard is not checking that closing brace is on line following the body

### ISSUE THREAD

**Issue by rtucek** [squizlabs/PHP_CodeSniffer#908]
[From php-fig](http://www.php-fig.org/psr/psr-2/#4-3-methods)

> Method names MUST NOT be declared with a space after the method name. The opening brace MUST go on its own line, and the closing brace MUST go on the next line following the body. _**There MUST NOT be a space after the opening parenthesis, and there MUST NOT be a space before the closing parenthesis.**_

So based on the definition above a function/method like this:

```php
public function some_function()
{


    // Some code here
   



    // Some code there

    return $something;


}
```

should be converted to this (e.g. stripping spaces and linebreaks in the body between the parenthesis and the first/last occurrence of a code block/comment).

```php
public function some_function()
{
    // Some code here
   



    // Some code there

    return $something;
}
```

**Comment by aik099** [comment]
1. You haven't specified into what that method is actually converted to.
2. Are all items from php-fig quote reported as as errors/warnings during `phpcs` run?
3. Are all items from php-fig quote fixed by PHP_CodeSniffer during `phpcbf` run?

**Comment by rtucek** [comment]
Hi @aik099 

1. I did:

> stripping spaces and linebreaks in the body between the parenthesis and the first/last occurrence of a code block/comment

plus providing an example.

2 & 3. I'm just referring to the bold/italic formatted text.
There are no warnings/errors returned by `phpcs` and no fixes done by `phpcbf` in context of the emphasized sentence above. I believe there is no implementation (this is what I meant by 'not fully honoring').

**Comment by gsherwood** [comment]
I think you are half right here.

The bold text refers to the parenthesis, not the braces. It is trying to ban code like this:
```php
<?php
public function some_function( $foo, $bar )
{
}
```
With that code, you get these errors:
```
 2 | ERROR | [x] Expected 0 spaces between opening bracket and argument "$foo"; 1 found
 2 | ERROR | [x] Expected 0 spaces between argument "$bar" and closing bracket; 1 found
```
So the rules inside the bold text are already checked, but by a different sniff.

There is also nothing in that rule that says that the body of the method must go on the line directly after the opening brace. In PSR2, you can have as much whitespace at the top of your functions as you'd like. Yes, I think this is crazy. No, I didn't make the rules.

But where I think you are very right is that the rule says that you can't have empty lines at the end of a function because the closing brace must go on the next line after the body. I think that is pretty clear and I should add both a check and an auto-fix for that.

Thanks for reporting it.

**Comment by gsherwood** [comment]
I finally got around to adding a sniff to check and fix extra blank lines at the end of functions/methods/closures.
